src/libs/Animals/Animals.cpp

```diff
@@ -4,16 +4,13 @@
 
 ANIMALS::ANIMALS() : seagulls(nullptr), fishSchools(nullptr), butterflies(nullptr)
 {
-    seagulls = new TSeagulls();
-    fishSchools = new TFishSchools();
-    butterflies = new TButterflies();
+    seagulls = std::make_shared<TSeagulls>();
+    fishSchools = std::make_shared<TFishSchools>();
+    butterflies = std::make_shared<TButterflies>();
 }
 
 ANIMALS::~ANIMALS()
 {
-    delete seagulls;
```

Comment:
Empty body destructor, might as well replace it with virtual ~ANIMALS() = default; in the header file.